File: src/main/java/com/yu/iotplatform/entity/MqttPublishLog.java
package com.yu.iotplatform.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.OffsetDateTime;

@Data
@TableName("\"mqtt_publish_log\"")
public class MqttPublishLog {
	@TableId(type = IdType.AUTO)
	private Long id;

	@TableField("topic")
	private String topic;

	@TableField("payload")
	private String payload;

	@TableField("qos")
	private Integer qos;

	@TableField("retained")
	private Boolean retained;

	@TableField("client_id")
	private String clientId;

	@TableField("broker_url")
	private String brokerUrl;

	@TableField(value = "create_time", fill = FieldFill.INSERT)
	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ssXXX", timezone = "GMT+8")
	private OffsetDateTime createTime;
}

File: entity/mqtt_publish_log.go
package entity

import (
	"github.com/yu/iot-platform-go/common"
)

// MqttPublishLog MQTT发布日志，对应表 mqtt_publish_log
type MqttPublishLog struct {
	ID         uint            `gorm:"primaryKey;autoIncrement" json:"id"`
	Topic      string          `gorm:"column:topic;size:255;not null" json:"topic"`
	Payload    string          `gorm:"column:payload;type:text" json:"payload"`
	Qos        int             `gorm:"column:qos;not null;default:0" json:"qos"`
	Retained   bool            `gorm:"column:retained;not null;default:false" json:"retained"`
	ClientID   string          `gorm:"column:client_id;size:128" json:"clientId"`
	BrokerURL  string          `gorm:"column:broker_url;size:255" json:"brokerUrl"`
	CreateTime common.DateTime `gorm:"column:create_time" json:"createTime"`
}

// TableName 指定表名
func (MqttPublishLog) TableName() string {
	return "mqtt_publish_log"
}
